source/audio/soundSplit.py

```diff
@@ -192,3 +209,45 @@ def toggleSoundSplitState() -> None:
 			"one of audio sessions is either mono, or has more than 2 audio channels."
 		)
 		ui.message(msg)
+
+
+@dataclass(unsafe_hash=True)
+class VolumeRestorer(AudioSessionEvents):
+	pid: int
+	audioSession: AudioSession
+
+	def on_state_changed(self, new_state: str, new_state_id: int):
+		if new_state == "Expired":
+			# For some reason restoring volume doesn't work in this thread, so scheduling in the main thread.
+			core.callLater(0, self.restoreVolume)
+
+	def restoreVolume(self):
+		# Application connected to this audio session is terminating. Restore its volume.
+		try:
+			channelVolume = self.audioSession.channelAudioVolume()
+			channelCount = channelVolume.GetChannelCount()
+			if channelCount != 2:
+				log.warning(
+					f"Audio session for pid {self.pid} has {channelCount} channels instead of 2 - cannot set volume!"
+				)
+				return
+			channelVolume.SetChannelVolume(0, 1.0, None)

```

Comment:
Why should the volume be 1.0? What if the application itself lowered the volume of its channels?
I think the volume of both channels should be restored to the volume of the active channel. SO if soundsplit was active with NVDA left, application right and the application lowered its volume to 0.5, both channels should be set to 0.5 instead of 1.0.